modules/apps/notification/notification-api/src/main/java/com/liferay/notification/type/BaseNotificationType.java

```diff
@@ -55,13 +57,34 @@
 import java.util.Objects;
 import java.util.Set;
 
+import javax.servlet.http.HttpServletRequest;
+
 import org.osgi.service.component.annotations.Reference;
 
 /**
  * @author Feliphe Marinho
  */
 public abstract class BaseNotificationType implements NotificationType {
 
+	public static void prepareGeneralVariablesTemplate(
```

Comment:
Move this method (and rename it to `contribute`) to a new class inside `notification-api` called `NotificationTemplateTermValuesContributor`. Check the contributors patterns inside portal.